SCons/Action.py

```diff
@@ -1256,8 +1254,7 @@ def __str__(self) -> str:
         try:
             env = self.presub_env
         except AttributeError:
-            env = None
-        if env is None:
+            import SCons.Defaults  # pylint: disable=import-outside-toplevel
             env = SCons.Defaults.DefaultEnvironment()
```

Comment:
Is throwing the exception the only path leading to env=None?

Reply:
yeah the `try` is the first line of the method and it takes no args (well, `self`...)

Reply:
Nevermind, I'll undo this bit.  `self.presub_env` may well be `None`, because of this brilliance.
```
        self.presub_env = env                                                  
        lines = str(self).split('\n')                                          
        self.presub_env = None      # don't need this any more  
```